### ISSUE TITLE
Add a method that processes all events currently inside Bloc's sink at once

### ISSUE THREAD

**Issue by Code-Sage** [felangel/bloc#1983]
**Is your feature request related to a problem? Please describe.**
I've been trying to find a method that I can use, and, after calling it, be care-free about where and/or which Flutter code may access the Bloc's state. I don't want to use close( ) since that'll just make the bloc incapable of further use. I've ran into situations where when Bloc's state is accessed on a later screen(widget) in the app, I don't get the state that the last-added event was supposed to produce.

**Describe the solution you'd like**
Add a method that forces processing of all events at once, synchronously, so that after its called, the developer can be rest assured that the state I get is the latest one.

**Comment by felangel** [comment]
Hi @Code-Sage 👋 
Thanks for opening an issue!

Can you please provide more context around the use-case? I'm not sure I understand the request.

The UI should not know/care about the rate at which the bloc is processing events. Also, the UI always gets the latest state. If you are asking for a way to wait until there are no more events being processed that is not feasible unless you close the bloc because an event can be added at any point in time (which would cause you to wait endlessly). In general, the UI should be prepared to handle any state at any time.

Hope that helps 👍

**Comment by Code-Sage** [comment]
Okay, so, there are 2 screens, I'll go through each of their flow one by one:
Screen 1: Phone number input
- user selects dial code from a drop-down.
- user types the rest of the phone number in a text field.
- while the user is typing of selecting a dial code, my bloc emits PhoneNumEditingState, the currently typed value of phone number and dial code is passed along with the PhoneNumEditingEvents added to the bloc.
- and once the user is done and hits continue button at the bottom, the bloc emits PhoneNumAcceptedState, and the final values of dial code and phone number are passed in the PhoneNumSubmittedEvent.
- **Note** that having those 2 states for the same screen may seem confusing, but I did it so that even if the user comes back to this screen from one of the later screens, the user could edit the phone number and the updated value is stored in the bloc's state. [ I did it because the bloc won't update the data member if the state being yield is of the same class i.e. adding multiple events of PhoneNumSubmittedEvent won't store the updated value of phone number in the state. ]

Screen 2: OTP Verification (just the concept of the screen, no OTP currently being done)
- the phone number that was input from Screen 1 is displayed, to be OTP verified
- for now, the user simply hits continue button and the phone number (dial code + rest of the number) is saved to FlutterSecureStorage (so that later on, the same number only would be identified as verified, and new phone numbers would have to be verified from scratch).

Now, I need the dial code and number stored in PhoneNumAcceptedState, which is a result of user finalizing the phone number and hitting continue, but when I access the state, I get PhoneNumEditingState. Now, since there a lot of PhoneNumEditingStates being emitted for each typing during input on Screen 1, how can I be sure that I'm getting the right value? If I get PhoneNumAcceptedState, then I can be assured that I have the finalized number. But I can't be sure if the state I get is PhoneNumEditingState.
